### PR TITLE
Olha/UI/update/application page

### PR DESCRIPTION
**Pull request by olhasashchuk**
# Enhance Application Page and Responsiveness

## Description of Changes:

This pull request includes several updates to improve the application page functionality, ensure better responsiveness across different devices, and streamline the application details display process. Additionally, it introduces mobile responsiveness, error handling, and better user experience on different devices. The following changes have been implemented:

### Application Component Updates:
1. **Integrated Error and Loading States**:
   - Improved user experience by handling loading and error states, ensuring the application only displays details (header, status, management panels) when the data has been successfully fetched.
   
2. **Components Display**:
   - The components (header, status, job details, etc.) are displayed only when the application data is available, improving performance and user experience.

### Mobile Responsiveness:
1. **Responsive Design Implementation**:
   - Introduced the `useIsMobile` hook to detect the device type (mobile or desktop).
   - Wrapped key components in a `ResponsiveWrapper` to adjust their layout based on screen size, ensuring the app works well across mobile and desktop devices.

### Date Picker Integration:
1. **DatePickerField Component**:
   - Added a custom `DatePickerField` to handle date fields with built-in validation.
   
2. **Updated AddApplicationForm**:
   - Integrated the `DatePickerField` component into the `AddApplicationForm` for selecting application and deadline dates, with full validation handling via `react-hook-form`.

### Localization Support:
1. **English Locale for Date Pickers**:
   - Configured date pickers to use the English locale (`enGB`) to improve internationalization and make the app suitable for different locales.

### Form Validation:
1. **Utilized `react-hook-form`**:
   - Improved form validation by integrating `react-hook-form` to manage form control, validation, and error handling, ensuring data is properly validated before submission.

### Manage Panel Structure:
1. **Components Added**:
   - **`ApplicationHeader`**: Displays the job title, company name, and location with a button to add a salary if missing.
   - **`ControlButton`**: A component for managing the application with buttons for editing, deleting, and navigating.
   - **`JobInfo`**: Displays job details such as job description, application date, and deadline date.
   - **`Contacts`**: A section for managing application contacts, with a card layout for each contact.
   - **`ManagePanel`**: The main panel structure that includes tabs for managing the application:
     - **Job**: Displays job-related information.
     - **Interview**: Placeholder for interview status and notes.
     - **Documents**: Placeholder for important documents.
     - **Notes**: Placeholder for application notes.
     - **Contacts**: Displays contact information.

## Related Issues:
- No related issues.

## How to Test:
1. Verify that the application displays correctly on both desktop and mobile devices, adjusting layout based on screen size.
2. Test that the application data only displays once the data is successfully fetched.
3. Ensure that the `DatePickerField` works correctly for selecting dates, and that form validation is functional.
4. Confirm that error and loading states are correctly handled during data fetching.
5. Check that all components (Job Info, Contacts, Notes, etc.) are displayed correctly.

## Screenshots:
- Video Overview 

https://github.com/user-attachments/assets/759d2335-f4b5-4b48-b803-f0669a779b0b

Screenshot_4
Screenshot_3
Screenshot_2
Screenshot_1
Screenshot_5

## Notes:
- This pull request focuses on the structure, design, and responsiveness of the application management system. Functionality and data handling will be implemented in future pull requests.

### CODE REVIEW

File: frontend/app/applications/[id]/ApplicationHeader.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import ManagePanel from './ManagePanel'
+import { useIsMobile } from '@/app/hooks/useIsMobile'
+
+export default function ApplicationHeader({ application }) {
+  const isMobile = useIsMobile()
+  return (
+    <Box>
+      <Typography
+        variant={isMobile ? 'h6' : 'h5'}
+        component="div"
```
Comment (Natata08) [inline frontend/app/applications/[id]/ApplicationHeader.js:15]: What do you think about making this component "h2" for semantics?

File: frontend/app/applications/[id]/ApplicationHeader.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
+'use client'
+import { Typography, Stack, Link, Box, Button } from '@mui/material'
+import ControlButton from './ControlButton'
+import StatusPanel from './StatusPanel'
+import AddIcon from '@mui/icons-material/Add'
+import ManagePanel from './ManagePanel'
```
Comment (Natata08) [inline frontend/app/applications/[id]/ApplicationHeader.js:6]: Could you please delete unused components: ControlButton, StatusPanel, ManagePanel

File: frontend/app/applications/[id]/ApplicationHeader.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+          component="div"
+          sx={{
+            pb: 2,
+            color: 'secondary.main',
+            fontSize: { xs: '1rem', sm: '1.5rem' },
+          }}
+        >
+          {application.salary !== undefined && application.salary !== null ? (
+            application.salary === 0 ? (
+              'Unpaid'
```
Comment (Natata08) [inline frontend/app/applications/[id]/ApplicationHeader.js:63]: Cool!

File: frontend/app/applications/[id]/ApplicationHeader.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+              '&:hover': { textDecoration: 'underline' },
+              wordBreak: 'break-word',
+            }}
+          >
+            {application.website}
+          </Link>
+        </Box>
+        <Typography
+          variant="body1"
+          component="div"
```
Comment (Natata08) [inline frontend/app/applications/[id]/ApplicationHeader.js:54]: Maybe a paragraph would be better?

File: frontend/app/applications/[id]/Contacts.js [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+          justifyContent: 'start',
+          alignItems: { xs: 'stretch', sm: 'center' },
+          pb: 2,
+        }}
+      >
+        <Card sx={{ minWidth: 275 }}>
+          <CardContent>
+            <Typography
+              variant="h5"
+              component="div"
```
Comment (Natata08) [inline frontend/app/applications/[id]/Contacts.js:32]: Maybe component is also h5

File: frontend/app/applications/[id]/ControlButton.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+  const handleOpenModal = () => {
+    setOpenModal(true)
+  }
+
+  const handleCloseModal = () => {
+    setOpenModal(false)
+  }
+
+  return (
+    <>
```
Comment (Natata08) [inline frontend/app/applications/[id]/ControlButton.js:26]: Unnecessary fragments

**Review by Natata08 — CHANGES_REQUESTED** [review]
Perfect job! I left just minor comments

File: frontend/app/applications/[id]/StatusPanel.js [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+
+const StatusPanel = ({ application }) => {
+  const isMobile = useIsMobile()
+  console.log(application.status)
+
+  const storedStatus =
+    typeof window !== 'undefined' ? localStorage.getItem('status') : null
+  const [status, setStatus] = useState(storedStatus || 'Saved')
+
+  const statuses = [
```
[the remark this answers is not in the document]
Comment (olhasashchuk) [reply frontend/app/applications/[id]/StatusPanel.js:30]: I will fix it in the future PR. And I want to use fetchStatuses.

**Review by Natata08 — APPROVED** [review]

**Review by myselfankitaa — APPROVED** [review]

**Pull request merged**